jangaroo-maven/jangaroo-maven-plugin/src/main/java/net/jangaroo/jooc/mvnplugin/WorkspaceConverterMojo.java

```diff
@@ -184,369 +185,374 @@ public void execute() throws MojoExecutionException, MojoFailureException {
-    try {
-      synchronized (lock) {
-        rootPackageJson.readPackageJson();
-        moduleMappings.entrySet().stream()
-                .map(entry -> {
-                  if (ModuleType.IGNORE == entry.getValue().getModuleType()) {
-                    return null;
-                  } else if (ModuleType.AGGREGATOR == entry.getValue().getModuleType() && !isAggregatorToConvert(entry.getValue().getData())) {
-                    return null;
-                  } else {
-                    return getPackageFolderName(entry.getKey());
-                  }
-                })
-                .filter(Objects::nonNull)
-                .forEach(rootPackageJson::addWorkspace);
-
-        rootPackageJson.writePackageJson();
-
-        File eslintRcFile = Paths.get(convertedWorkspaceTarget + "/.eslintrc.js").toFile();
-        FileUtils.copyInputStreamToFile(getClass().getResourceAsStream("/net/jangaroo/jooc/mvnplugin/.eslintrc.js"), eslintRcFile);
-
-        Map<String, Object> lernaJson = new LinkedHashMap<>();
-        lernaJson.put("useWorkspaces", true);
-        lernaJson.put("version", "0.0.0");
-        Map<String, Object> lernaCommandMap = new LinkedHashMap<>();
-        Map<String, Object> lernaRunMap = new LinkedHashMap<>();
-        lernaRunMap.put("stream", true);
-        lernaCommandMap.put("run", lernaRunMap);
-        lernaJson.put("command", lernaCommandMap);
-
-        FileUtils.write(new File(convertedWorkspaceTarget + "/lerna.json"), objectMapper.writerWithDefaultPrettyPrinter().writeValueAsString(lernaJson).concat("\n"));
+    if (!new File(convertedWorkspaceTarget).exists()) {
+      new File(convertedWorkspaceTarget).mkdirs();
+    }
+    synchronized (session.getLocalRepository()) {
+      RootPackageJson rootPackageJson = new RootPackageJson(objectMapper, convertedWorkspaceTarget);
+      List<String> workspaces = moduleMappings.entrySet().stream()
+              .map(entry -> {
+                if (ModuleType.IGNORE == entry.getValue().getModuleType()) {
+                  return null;
+                } else if (ModuleType.AGGREGATOR == entry.getValue().getModuleType() && !isAggregatorToConvert(entry.getValue().getData())) {
+                  return null;
+                } else {
+                  return getPackageFolderName(entry.getKey());
+                }
+              })
+              .filter(Objects::nonNull)
+              .collect(Collectors.toList());
+      try {
+        rootPackageJson.writePackageJson(workspaces);
+      } catch (IOException e) {
+        throw new MojoFailureException(e.getMessage(), e.getCause());
       }
+    }
 
-      List<String> excludePaths = new ArrayList<>();
-      if (!optionalPackage.isPresent()) {
-        logger.warn("Package was null");
-        return;
+    synchronized (session.getLocalRepository()) {
```

Comment:
Why do you also synchronize here? As discussed, it should be possible to do just try to copy without replacing and ignore any "file already exists" exceptions. You could even do that for every single file, so that all missing files would be updated (see comment below).